aws_route53recoverycontrolconfig_routing_control panics with nil pointer dereference on create

### Terraform and AWS Provider Version

```console
Terraform version: 1.13.5

Provider version: 6.36.0 and 6.37.0
```

### Affected Resource(s) or Data Source(s)

aws_route53recoverycontrolconfig_routing_control

### Expected Behavior

Provider handles 409 ConflictException with a retry (as it does for other AWS services), or returns a proper error without crashing. 

### Actual Behavior

  1. All 12 resources are created in parallel
  2. The AWS API returns 409 ConflictException on concurrent mutations to the same cluster                                                 
  3. The provider dereferences a nil pointer while handling the error response, crashing the plugin process
  4. The crash propagates as Request cancelled to all other in-flight creates                                                              
  5. Some resources may have been successfully created in AWS but are absent from state     

### Relevant Error/Panic Output

```                                                                                                                                         
  panic: runtime error: invalid memory address or nil pointer dereference
                                                                                                                                           
  goroutine 111:                                                                                                                           
  github.com/hashicorp/terraform-provider-aws/internal/service/route53recoverycontrolconfig.resourceRoutingControlCreate(...)              
          .../route53recoverycontrolconfig/routing_control.go:78 +0x319  

```


### Sample Terraform Configuration

<details open>
<summary>Click to expand configuration</summary>

```hcl
locals {                                                                                                                                 
    control_panel_config = {
      "global-writer" = ["aurora-primary-us-e1", "aurora-primary-eu-c1", "aurora-primary-ap-s2"]                                           
      "us-east-1"     = ["aurora-secondary-us-e1", "aurora-secondary-eu-c1", "aurora-secondary-ap-s2"]                                     
      "eu-central-1"  = ["aurora-secondary-us-e1", "aurora-secondary-eu-c1", "aurora-secondary-ap-s2"]                                     
      "ap-southeast-2" = ["aurora-secondary-us-e1", "aurora-secondary-eu-c1", "aurora-secondary-ap-s2"]                                    
    }                                                                                                                                      
                                                                                                                                           
    routing_controls = merge([                                                                                                             
      for prefix, suffixes in local.control_panel_config : {                                                                               
        for suffix in suffixes : "${prefix}-${suffix}-rc" => "my-control-panel"                                                            
      }                                                                                                                                    
    ]...)                                                                                                                                  
  }

resource "aws_route53recoverycontrolconfig_routing_control" "this" {                                                                     
    for_each          = local.routing_controls  # 12 entries                                                                               
    name              = each.key                                                                                                           
    cluster_arn       = "<cluster_arn>"                                                                                                    
    control_panel_arn = aws_route53recoverycontrolconfig_control_panel.this[each.value].arn                                                
  }   
```
</details>


### Steps to Reproduce

Create an aws_elasticache_reserved_cache_node resource using provider v6.36.0 or  v6.37.0
Run terraform apply
All computed attributes return as unknown, causing the apply to fail

### Debug Logging

<details open>
<summary>Click to expand log output</summary>

```console

aws_route53recoverycontrolconfig_control_panel.this["resource"]: Creating...
aws_route53recoverycontrolconfig_control_panel.this["resource"]: Still creating... [00m10s elapsed]
aws_route53recoverycontrolconfig_control_panel.this["resource"]: Creation complete after 17s 
aws_route53recoverycontrolconfig_routing_control.this["global-writer-aurora-primary-eu-w1-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["us-east-1-aurora-secondary-eu-w1-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["eu-west-1-aurora-secondary-us-e1-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["eu-west-1-aurora-secondary-eu-w1-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["us-east-1-aurora-secondary-us-w2-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["us-west-2-aurora-secondary-eu-w1-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["us-west-2-aurora-secondary-us-e1-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["us-west-2-aurora-secondary-us-w2-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["eu-west-1-aurora-secondary-us-w2-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["us-east-1-aurora-secondary-us-e1-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["global-writer-aurora-primary-us-e1-rc"]: Creating...
aws_route53recoverycontrolconfig_routing_control.this["global-writer-aurora-primary-us-w2-rc"]: Creating...

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["eu-west-1-aurora-secondary-us-w2-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["eu-west-1-aurora-secondary-eu-w1-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["global-writer-aurora-primary-us-e1-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["us-east-1-aurora-secondary-us-w2-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["global-writer-aurora-primary-eu-w1-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["us-west-2-aurora-secondary-us-e1-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["us-east-1-aurora-secondary-us-e1-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["us-west-2-aurora-secondary-eu-w1-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["eu-west-1-aurora-secondary-us-e1-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["us-east-1-aurora-secondary-eu-w1-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["us-west-2-aurora-secondary-us-w2-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Error: Request cancelled

  with aws_route53recoverycontrolconfig_routing_control.this["global-writer-aurora-primary-us-w2-rc"],
  on routing_control.tf line 1, in resource "aws_route53recoverycontrolconfig_routing_control" "this":
   1: resource "aws_route53recoverycontrolconfig_routing_control" "this" {

The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Stack trace from the terraform-provider-aws_v6.37.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference


goroutine 213 :
github.com/hashicorp/terraform-provider-aws/internal/service/route53recoverycontrolconfig.resourceRoutingControlCreate({0x24872868, 0xc00b16a540}, 0xc00b1662d0, {0x2440df40, 0xc000d9d0e0})
        github.com/hashicorp/terraform-provider-aws/internal/service/route53recoverycontrolconfig/routing_control.go:78 +0x319
github.com/hashicorp/terraform-provider-aws/internal/provider/sdkv2.wrapResource.(*wrappedResource).create.interceptedCRUDHandler[...].func2(0xc00b1662d0, {0x2440df40, 0xc000d9d0e0})
        github.com/hashicorp/terraform-provider-aws/internal/provider/sdkv2/intercept.go:178 +0x5d5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x24872868?, {0x24872868?, 0xc00b103c50?}, 0xd?, {0x2440df40?, 0xc000d9d0e0?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.0/helper/schema/resource.go:844 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001463900, {0x24872868, 0xc00b103c50}, 0xc00b13c5b0, 0xc00b1661b0, {0x2440df40, 0xc000d9d0e0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.0/helper/schema/resource.go:980 +0xb33
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc005fc44f8, {0x24872868?, 0xc00b103b90?}, 0xc00b108eb0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.0/helper/schema/grpc_provider.go:1500 +0x1089
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc0077bc090, {0x24872868?, 0xc00b1038c0?}, 0xc00b108eb0)
        github.com/hashicorp/terraform-plugin-mux@v0.23.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x18d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0070c5400, {0x24872868?, 0xc00b26f5f0?}, 0xc00b204980)
        github.com/hashicorp/terraform-plugin-go@v0.31.0/tfprotov5/tf5server/server.go:944 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x242cef00, 0xc0070c5400}, {0x24872868, 0xc00b26f5f0}, 0xc00b204900, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.31.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:805 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc007930488, {0x24872868, 0xc00b26f560}, 0xc00b278ea0, 0xc00793ce10, 0x34084088, 0x0)
        google.golang.org/grpc@v1.79.2/server.go:1426 +0x1192
google.golang.org/grpc.(*Server).handleStream(0xc007930488, {0x24873278, 0xc000b54b60}, 0xc00b278ea0)
        google.golang.org/grpc@v1.79.2/server.go:1830 +0xdc6
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.79.2/server.go:1061 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 68
        google.golang.org/grpc@v1.79.2/server.go:1072 +0x11d

Error: The terraform-provider-aws_v6.37.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```
</details>


### GenAI / LLM Assisted Development

Claude Code

### Important Facts and References

When creating multiple aws_route53recoverycontrolconfig_routing_control resources in parallel, the Route53 Recovery Control Config API
  returns a 409 ConflictException for concurrent mutations on the same cluster. The provider does not handle this error response gracefully
   — instead it dereferences a nil pointer at routing_control.go:78, crashing the plugin process. This crash then causes all other
  in-flight gRPC ApplyResourceChange calls to be cancelled, resulting in 11 Request cancelled errors and 1 panic for a batch of 12         
  resources.                                                

  The same issue occurs on delete: parallel deletes return 409s, but the delete path handles them as proper errors rather than panicking   
  (no nil dereference on delete).
                                                                                                                                           
  Side effect: The API calls that were in-flight before the crash succeed in AWS — resources are created — but the provider crash means    
  Terraform never writes them to state, causing state drift.

Workaround                                                                                                                               
                                                            
  Split the for_each into sequential batches using depends_on:                                                                             
                                                            
  resource "aws_route53recoverycontrolconfig_routing_control" "global" {                                                                   
    for_each          = local.routing_controls_global                                                                                      
    name              = each.key                                                                                                           
    cluster_arn       = "<cluster_arn>"                                                                                                    
    control_panel_arn = aws_route53recoverycontrolconfig_control_panel.this[each.value].arn                                                
  }                                                                                                                                        
                                                                                                                                           
  resource "aws_route53recoverycontrolconfig_routing_control" "regional" {                                                                 
    for_each          = local.routing_controls_regional                                                                                    
    name              = each.key                                                                                                           
    cluster_arn       = "<cluster_arn>"                                                                                                    
    control_panel_arn = aws_route53recoverycontrolconfig_control_panel.this[each.value].arn                                                
                                                                                                                                           
    depends_on = [aws_route53recoverycontrolconfig_routing_control.global]                                                                 
  }

### Would you like to implement a fix?

No

--- a/.changelog/47038.txt
+++ b/.changelog/47038.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_route53recoverycontrolconfig_routing_control: Fix panic on concurrent creates when API returns ConflictException
+```

--- a/internal/service/route53recoverycontrolconfig/routing_control.go
+++ b/internal/service/route53recoverycontrolconfig/routing_control.go
@@ -75,12 +75,12 @@ func resourceRoutingControlCreate(ctx context.Context, d *schema.ResourceData, m
 	}
 
 	output, err := conn.CreateRoutingControl(ctx, input)
-	result := output.RoutingControl
 
 	if err != nil {
 		return sdkdiag.AppendErrorf(diags, "creating Route53 Recovery Control Config Routing Control: %s", err)
 	}
 
+	result := output.RoutingControl
 	if result == nil {
 		return sdkdiag.AppendErrorf(diags, "creating Route53 Recovery Control Config Routing Control: empty response")
 	}

